### PR TITLE
Improve usability of the request localization middleware #164

### PR DESCRIPTION
**Pull request by DamianEdwards**
#164 

Fixed up some out of date doc comments too.

@rynowak @ryanbrandenburg @Eilon

### CODE REVIEW

**Comment by hishamco** [comment]
I'm not sure if the issue #127 should take place here or will be ignored

File: src/Microsoft.AspNet.Localization/RequestLocalizationOptions.cs [hunk 1]
```diff
@@ -25,20 +29,40 @@ public RequestLocalizationOptions()
...
+        {
+            get
+            {
+                return _defaultRequestCulture;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException();
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNet.Localization/RequestLocalizationOptions.cs]: Should have the name of the property in the exception yeah?

File: src/Microsoft.AspNet.Localization/ApplicationBuilderExtensions.cs [hunk 2]
```diff
@@ -63,12 +33,7 @@ public static class ApplicationBuilderExtensions
                 throw new ArgumentNullException(nameof(options));
             }
 
-            if (defaultRequestCulture == null)
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNet.Localization/ApplicationBuilderExtensions.cs]: We checked that we didn't set the DefaultRequestCulture to null, but are we checking that it has ever been set anywhere or did I miss it?

Comment (DamianEdwards) [inline src/Microsoft.AspNet.Localization/ApplicationBuilderExtensions.cs]: It's initialized directly via the backing field https://github.com/aspnet/Localization/pull/170/files#diff-61f74221a354a19742c79b0c32c46423R15

File: src/Microsoft.AspNet.Localization/RequestLocalizationOptions.cs (hunk 1 above)
Comment (DamianEdwards) [inline src/Microsoft.AspNet.Localization/RequestLocalizationOptions.cs]: Yeah I actually wasn't sure what name to use here, but property name makes sense.

File: src/Microsoft.AspNet.Localization/ApplicationBuilderExtensions.cs (hunk 2 above)
Comment (ryanbrandenburg) [inline src/Microsoft.AspNet.Localization/ApplicationBuilderExtensions.cs]: So it is!

**Comment by DamianEdwards** [comment]
@hishamco that will be separate

File: src/Microsoft.AspNet.Localization/RequestLocalizationOptions.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Localization/RequestLocalizationOptions.cs]: This should be `nameof(value)`. That is the name of the parameter.

**Comment by ryanbrandenburg** [comment]
:shipit:

**Pull request merged**